yt/utilities/parallel_tools/parallel_analysis_interface.py

```diff
@@ -1157,9 +1199,9 @@ def alltoallv_array(self, send, total_size, offsets, sizes):
         dtr = send.dtype.itemsize / tmp_send.dtype.itemsize  # > 1
         roff = [off * dtr for off in offsets]
         rsize = [siz * dtr for siz in sizes]
-        tmp_recv = recv.view(self.__tocast)
+        tmp_recv = recv.view(transport_dtype)
```

Comment:
The motivation for getting rid of the cast to CHAR is so that we do not hit the limit imposed by MPI of 2**31 elements when communicating arrays as quickly.